### PR TITLE
feat: realize building unit val4

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

**Review by pgallik — APPROVED** [review]

File: src/BuildingRegistry/Building/Exceptions/BuildingUnitStatusPreventsRealizeBuildingUnitException.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace BuildingRegistry.Building.Exceptions
+{
+    public class BuildingUnitStatusPreventsBuildingUnitRealizationException : BuildingRegistryException
+    {
+        public BuildingUnitStatusPreventsBuildingUnitRealizationException(BuildingUnitStatus status)
+            : base($"Cannot realize buildingUnit with status '{status}'.") { }
```
Comment (pgallik) [inline src/BuildingRegistry/Building/Exceptions/BuildingUnitStatusPreventsRealizeBuildingUnitException.cs:6]: verwijder parameter constructor

Comment (emalfroy) [reply src/BuildingRegistry/Building/Exceptions/BuildingUnitStatusPreventsRealizeBuildingUnitException.cs]: done

File: src/BuildingRegistry.Api.BackOffice.Abstractions/Building/Validators/ValidationErrors.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@ public static class BuildingUnit
         {
             public const string BuildingUnitNotFound = "GebouweenheidNietGevonden";
             public const string BuildingUnitIsRemoved = "GebouweenheidVerwijderd";
+            public const string BuildingStatusNotInRealized = "GebouwStatusNietInGerealiseerd";
```
Comment (lvermeulen) [inline src/BuildingRegistry.Api.BackOffice.Abstractions/Building/Validators/ValidationErrors.cs:17]: "GebouwStatusNietGerealizeerd"

File: src/BuildingRegistry/Building/Exceptions/BuildingStatusPreventsRealizeBuildingUnitException.cs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+namespace BuildingRegistry.Building.Exceptions
+{
+    public class BuildingStatusPreventsBuildingUnitRealizationException : BuildingRegistryException
+    {}
```
Comment (lvermeulen) [inline src/BuildingRegistry/Building/Exceptions/BuildingStatusPreventsRealizeBuildingUnitException.cs:4]: add a space between the braces

File: src/BuildingRegistry/Building/Exceptions/BuildingUnitStatusPreventsRealizeBuildingUnitException.cs [hunk 4]
```diff
@@ -0,0 +1,5 @@
+namespace BuildingRegistry.Building.Exceptions
+{
+    public class BuildingUnitStatusPreventsBuildingUnitRealizationException : BuildingRegistryException
+    {}
```
Comment (lvermeulen) [inline src/BuildingRegistry/Building/Exceptions/BuildingUnitStatusPreventsRealizeBuildingUnitException.cs:4]: add a space between the braces

**Pull request merged**